File: src/main/resources/static/js/service/team-service.js
app.factory('teamService', function ($http) {

    function create(team, coachId) {
        return $http.post('api/team/' + coachId, team);
    }

    function findOne(teamId) {
        return $http.get('api/team/' + teamId);
    }

    function findByCoach(coachId) {
        return $http.get('api/team/coach/' + coachId);
    }

    function update(team) {
        return $http.put('api/team', team);
    }

    function deleteOne(teamId) {
        return $http.delete("api/team/" + teamId);
    }

    return {
        create: create,
        findOne: findOne,
        findByCoach: findByCoach,
        update: update,
        deleteOne: deleteOne
    }

});

File: src/main/java/br/com/poketeams/model/service/TeamService.java
package br.com.poketeams.model.service;

import br.com.poketeams.model.entity.Team;

import java.util.List;

public interface TeamService {

    Team create(Team team, Long coachId);

    Team findOne(Long teamId);

    List<Team> findAll(Long coachId);

    Team update(Team team);

    Team delete(Long teamId);

}
